From 0b6c6d816e5845827ee783782bd4d975069d5a3c Mon Sep 17 00:00:00 2001
From: Chen Zhang <zhangchen01@kylinos.cn>
Date: Thu, 18 Jun 2026 10:52:39 +0800
Subject: [PATCH] set: Abandon the use of the existing set gatt-db

When pairing and connecting a pair of earbuds, if you attempt to connect the
second earbud while one earbud is already connected, reusing the existing
set gatt-db will interrupt the GATT procedure. Subsequently, the attribute
values will no longer be read, and the ASE state will not switch. This will
eventually prevent the second earbud from creating a CIS, resulting in no sound
output from the later-connected earbud during music playback.

Signed-off-by: Zhang Chen <zhangchen01@kylinos.cn>
---
 src/set.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/set.c b/src/set.c
index 4ca2f78c37..0f2e9613c0 100644
--- a/src/set.c
+++ b/src/set.c
@@ -282,20 +282,6 @@ static void foreach_rsi(void *data, void *user_data)
 	if (memcmp(ad->data, res, sizeof(res)))
 		return;
 
-	/* Attempt to use existing gatt_db from set if device has never been
-	 * connected before.
-	 *
-	 * If dbs don't really match bt_gatt_client will attempt to rediscover
-	 * the ranges that don't match.
-	 */
-	if (gatt_db_isempty(btd_device_get_gatt_db(set->device))) {
-		struct btd_device *device;
-
-		device = queue_get_entries(set->devices)->data;
-		btd_device_set_gatt_db(set->device,
-					btd_device_get_gatt_db(device));
-	}
-
 	device_connect_le(set->device);
 }
 
